### ISSUE TITLE
Thermal monitor requires manual reload in every plasmashell restart

### ISSUE THREAD

**Issue by ongun-kanat** [kotelnik/plasma-applet-thermal-monitor#16]
When I start plasmashell thermal monitor does not automaticaly update itself. I have to right click and select reload every time I restarted plasmashell.

I'm using plasma 5.6 on Arch Linux

**Comment by kotelnik** [comment]
Yes, I'm now also noticing this since I've updated my Arch... Thanks! Will try to fix that soon.

**Comment by BilyakA** [comment]
4.1.12-gentoo with kde-plasma/plasma-meta-5.6.1 is here and no such problems.
Using last version of Thermal Monitor form Plasma Add-on installer

**Comment by kotelnik** [comment]
Thanks for info. There really is a problem with property initialization - I believe I've repaired that in latest commits. Please re-test from git and let me know. If it is fine I'll create a new tag and post a .plasmoid on kde-apps. Thanks!

**Comment by ongun-kanat** [comment]
Looks like no, it hasn't been resolved. I reinstalled from AUR git package.

**Comment by kotelnik** [comment]
Thanks! I looked into the problem and it looks like it is a general problem of initialization of array and map properties in plasma widgets. They get re-initialized **several** times when plasma is starting but also e.g. when external monitor is connected / disconnected. It can (and will) be repaired on my (widget) side. But I consider it a bug in plasma 5.6 (or Qt 5.6, I'm not sure which one's fault is it) so I'll report it elsewhere.

Please wait for my second attempt to repair this issue, I'll do that soon and get back to You.

**Comment by kotelnik** [comment]
Committed second attempt to fix this. Please try again :).

**Comment by wolfyrion** [comment]
it seems that the one from AUR is working fine

**Comment by kotelnik** [comment]
OK, thanks a lot, closing and making a tag.

**Comment by mamrhein** [comment]
I'm running a laptop and an Intel NUC, both with KDE Frameworks 5.24 and KDE Plasma 5.7.1.
While on the laptop Thermal Monitor is displaying the temperature instantly after start, on the NUC I have to reload the temperature sources each time.

**Comment by brigadier** [comment]
not resolved in KDE Plasma 5.7 from gentoo

**Comment by kotelnik** [comment]
Alright, thanks for reports, I'll investigate further.

**Comment by mamrhein** [comment]
With KDE Frameworks 5.27 and KDE Plasma 5.8.1 it does work on my Intel NUC too.
Thanks!

**Comment by glaubersm** [comment]
Here cpu temperature is not shown immediately after login. I need to reload thermal monitor manually.

thermal monitor 1.2.9
Operating System: Arch Linux 
KDE Plasma Version: 5.14.4
Qt Version: 5.12.0 RC2
KDE Frameworks Version: 5.52.0
Kernel Version: 4.19.4-arch1-1-ARCH

**Comment by Termuellinator** [comment]
I've now got the same problem since the last manjaro-update (see: [here](https://gist.githubusercontent.com/philmmanjaro/d3dbcd268c5db63ae3c146f04dab7eb1/raw/99bf95b4b747358a2f4405acffa50ab09f72279a/boxit-stable-2018-12-31.txt) )
It worked flawless in the past.
"sensors" in terminal shows the correct values, so it seems to be an issue with the initialisation of the widget again?

thermal monitor 1.2.8 / KDE 5.53.0 / Plasma 5.14.4 / Qt 5.12.0 on Kernel 4.20.0

**Comment by mamrhein** [comment]
After upgrading Qt and KDE, I have the same problem again. After start of Plasma CPU and MB temperature are not show, while disk temperature is.
Qt 5.12 / KDE Frameworks 5.54 / Plasma 5.14.4 / Thermal Monitor 1.2.8

**Comment by roachsinai** [comment]
Who care about this problem maybe could goto #53 to check the reason(not sure).

**Comment by artemanufrij** [comment]
Same here since a couple of weeks

**Comment by andreimarinescu** [comment]
yeah, having the same issue. did anyone figure out a workaround yet?

**Comment by roachsinai** [comment]
@andreimarinescu @artemanufrij #54 check it!

**Comment by artemanufrij** [comment]
@roachsinai thank you for your feedback...

**Comment by andreimarinescu** [comment]
@roachsinai I cloned your fork of the repo (the one used for the PR), followed the installation instructions and did a reboot. I still need to "Reload temperature sources" manually in order for the CPU temp to show up. Are there any other steps needed?

**Comment by roachsinai** [comment]
Maybe you should read this issue #53 for some details.

**Comment by veganvelociraptor** [comment]
The problem still persists for me on two different computers. One running Arch and the other one Manjaro.

**Comment by wolfyrion** [comment]
> The problem still persists for me on two different computers. One running Arch and the other one Manjaro.

for people who still have problems , simple solution

go -->

https://raw.githubusercontent.com/kotelnik/plasma-applet-thermal-monitor/b389b886a7ee759d06c9eb0e6a3b6abbea400035/package/contents/ui/main.qml 

Copy Everything

edit

/home/your username/.local/share/plasma/plasmoids/org.kde.thermalMonitor/contents/ui/main.qml

Paste Everything

Save, Restart and you are done

**Comment by veganvelociraptor** [comment]
Thanks @wolfyrion! That worked perfectly!

Do you know the reason why this is not already implemented as default?

**Comment by africsnail** [comment]
> Thanks @wolfyrion! That worked perfectly!
> 
> Do you know the reason why this is not already implemented as default?

Still isn't. At least on Manjaro
But the quick fix mentioned above solved it :)

**Comment by nagelp** [comment]
I also still have this problem on Ubuntu Linux, and the above-mentioned quick fix is working. Would I have gotten the fix if I had deleted the .local/share/plasma directory for my user? Don't want to try it now...

**Comment by tomchiverton** [comment]
> https://raw.githubusercontent.com/kotelnik/plasma-applet-thermal-monitor/b389b886a7ee759d06c9eb0e6a3b6abbea400035/package/contents/ui/main.qml

Fixed here, Ubuntu 20.04, Intel x86_64. The diff for that is just :

```
--- /home/falken//.local/share/plasma/plasmoids/org.kde.thermalMonitor/contents/ui/main.qml     2020-07-31 17:28:25.498982256 +0100
+++ /home/falken//.local/share/plasma/plasmoids/org.kde.thermalMonitor/contents/ui/main.qml.dist        2020-07-31 17:27:56.842175969 +0100
@@ -190,10 +190,6 @@
 
         temperatureModel.clear()
 
-        if (!systemmonitorAvailableSources) {
-            systemmonitorAvailableSources = []
-        }
-
         if (!systemmonitorSourcesToAdd) {
             systemmonitorSourcesToAdd = []
         }
@@ -406,4 +402,3 @@
     }
 
 }
```
